bin/tessel-2.js

```diff
@@ -77,42 +79,50 @@ function makeCommand(commandName) {
       choices: [true, false],
       abbr: 'o',
       help: 'Enable or disable writing command output to stdout/stderr. Useful for CLI API consumers.'
+    })
+    .option('loglevel', {
+      default: 'basic',
+      choices: ['trace', 'debug', 'basic', 'info', 'http', 'warn', 'error'],
+      help: 'Set the loglevel.',
     });
 }
 
 function callControllerWith(methodName, opts) {
+  log.spinner.stop();
   return controller[methodName](opts)
     .then(module.exports.closeSuccessfulCommand, module.exports.closeFailedCommand);
 }
 
-function callControllerCallback(methodName) {
-  return function(opts) {
-    return callControllerWith(methodName, opts);
-  };
-}
-
 parser.command('install-drivers')
-  .callback(function() {
+  .callback(options => {
+    if (options.loglevel !== log.level()) {
+      log.level(options.loglevel);
+    }
```

Comment:
This is implemented for all `callback(...)`. Whatever args handling we replace nom-nom with should include a "global" callback.